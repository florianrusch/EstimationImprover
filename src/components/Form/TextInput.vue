<template>
	<div class="flex items-center border-b border-teal-500 py-2">
		<input
			class="bg-transparent border-none mr-3 px-2 leading-tight focus:outline-none"
			type="text"
			v-model="inputValue"
			@focusout="onFocusOut"
		/>
	</div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue'

export default defineComponent({
	props: {
		value: {
			type: String,
			required: false,
		},
	},

	emits: ['change'],

	setup(props, { emit }) {
		const inputValue = ref(props.value)

		// eslint-disable-next-line @typescript-eslint/no-explicit-any
		const onFocusOut = (event: any) => {
			emit('change', event.target.value)
		}
		return {
			onFocusOut,
			inputValue,
		}
	},
})
</script>
