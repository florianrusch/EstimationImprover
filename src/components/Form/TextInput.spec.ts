import { DOMWrapper, mount } from '@vue/test-utils'
import TextInput from './TextInput.vue'

describe('# TextInput', () => {
	it('emits change event with new value', async () => {
		// given
		const wrapper = mount(TextInput, {
			props: { value: '' },
		})
		const inputElement = wrapper.find('input') as DOMWrapper<HTMLInputElement>
		const newValue = 'some texts ...'

		// when
		inputElement.element.value = newValue
		await inputElement.trigger('focusout')

		// then
		const events = wrapper.emitted('change')
		expect(events).toHaveLength(1)
		expect(events?.[0]).toEqual([newValue])
	})

	it('renders correctly w/o init value', () => {
		// given
		const wrapper = mount(TextInput)

		// when
		// then
		expect(wrapper.element).toMatchSnapshot()
	})

	it('renders correctly w/ init value', () => {
		// given
		const wrapper = mount(TextInput, {
			props: { value: 'Test value' },
		})

		// when
		// then
		expect(wrapper.element).toMatchSnapshot()
	})
})
