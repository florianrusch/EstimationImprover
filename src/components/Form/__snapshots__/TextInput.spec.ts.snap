// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# TextInput renders correctly w/ init value 1`] = `
<div
  class="flex items-center border-b border-teal-500 py-2"
>
  <input
    class="bg-transparent border-none mr-3 px-2 leading-tight focus:outline-none"
    type="text"
  />
</div>
`;

exports[`# TextInput renders correctly w/o init value 1`] = `
<div
  class="flex items-center border-b border-teal-500 py-2"
>
  <input
    class="bg-transparent border-none mr-3 px-2 leading-tight focus:outline-none"
    type="text"
  />
</div>
`;
